### PR TITLE
Skip project not assigned mail

### PR DESCRIPTION
**Pull request by tejaszambre**
- Added new field as unassigned_project in employee_details model default true
- Added send unassigned project mail button in employee details template
- Added a check before sending project not assigned mail to check unassigned_project flag and if the flag is set to false the mail will not be sent
- Modified test-cases for the above.

### CODE REVIEW

**Comment by tejaszambre** [comment]
[Skipping 'Project not assigned' mail](https://trello.com/c/rcizyYYF/21-skipping-project-not-assigned-mail)

File: app/models/employee_detail.rb [hunk 1]
```diff
@@ -49,6 +50,6 @@ def add_rejected_leave(number_of_days)
   def assign_new_usa_id?
     # For Sachin's record we are keeping his employee ID as it is
     # For all other USA employees ID must be above 9000
-    employee_id.nil? and location == "Plano" ? true : false
+    employee_id.nil? and location == 'Plano' ? true : false
```
Comment (Sameer-Tilak) [inline app/models/employee_detail.rb:53]: There is separate method to assign emp_id for different criteria as Consultant, Pune, Banglore, Plano etc. Why this duplicate method. I think we need to delete this and use one common method.

File: app/views/users/_employee_detail.html.haml [hunk 2]
```diff
@@ -7,9 +7,16 @@
...
-    = e.label "Is Billable"
-    .make-switch{"data-on" => "success", "data-off" => "warning", "data-on-label" => "Yes", "data-off-label" => "No"}
+    = e.label 'Is Billable'
+    .make-switch{'data-on': 'success', 'data-off': 'warning', 'data-on-label': 'Yes', 'data-off-label': 'No'}
       = e.check_box :is_billable, {}, 'Yes', 'No'
+    %br
+    %br
+    = e.label 'Send Project Not Assigned Mail'
+    .make-switch{'data-on': 'success', 'data-off': 'warning', 'data-on-label': 'Yes', 'data-off-label': 'No'}
+      = e.check_box :unassigned_project, {}, 'Yes', 'No'
```
Comment (Sameer-Tilak) [inline app/views/users/_employee_detail.html.haml:17]: Meaning of field name does not reflect correct use of it. I think this name should be "mail_for_unassigned_project" with default value as "Yes"

Comment (Ashish944) [reply app/views/users/_employee_detail.html.haml:17]: Okay will update the name

File: app/models/employee_detail.rb (hunk 1 above)
Comment (Ashish944) [reply app/models/employee_detail.rb:53]: yes it was used earlier but afterwards its validation was commented. I'll remove it.

File: app/models/employee_detail.rb [hunk 3]
```diff
@@ -11,13 +11,13 @@ class EmployeeDetail
   field :available_leaves, type: Integer, default: 0
   field :description
   field :is_billable, type: Boolean, default: false
+  field :skip_unassigned_project_ts_mail, type: Boolean, default: true
```
Comment (Sameer-Tilak) [inline app/models/employee_detail.rb:14]: You have renamed the variable name but did not change the default value. Per the change in the name of variable default value should be 'NO'.

Comment (tejaszambre) [reply app/models/employee_detail.rb:14]: yes okay, will change the default value to false and do necessary changes.

**Comment by Sameer-Tilak** [comment]
LGTM

**Pull request merged**